File: python/231-power-of-two.py
class Solution:
    def isPowerOfTwo(self, n: int) -> bool:
        exponent = 0
        base = 1
        while base <= n:
            if base == n:
                return True
            else:
                exponent += 1
                base = 2 ** exponent
        
        return False

print(Solution().isPowerOfTwo(2 ** 58))

File: typescript/src/231-power-of-two.ts
function isPowerOfTwo(n: number): boolean {
  let exponent = 0;
  let base = 1;

  while (base <= n) {
    if (base == n) {
      return true;
    } else {
      exponent += 1;
      base = 2 ** exponent;
    }
  }

  return false;
}

console.log(isPowerOfTwo(2))